src/components/Swap/Summary/index.tsx

```diff
@@ -91,13 +89,44 @@ interface SummaryDialogProps {
 }
 
 export function SummaryDialog(props: SummaryDialogProps) {
+  const [ackPriceImpact, setAckPriceImpact] = useState(false)
+  const [showSpeedbump, setShowSpeedbump] = useState(props.impact?.warning === 'error')
+
+  const onAcknowledgePriceImpact = useCallback(() => {
+    setAckPriceImpact(true)
+    setShowSpeedbump(false)
+  }, [])
+
+  const onAcknowledgeNewTrade = useCallback(() => {
+    if (!showSpeedbump && !ackPriceImpact && props.impact?.warning === 'error') {
+      setShowSpeedbump(true)
+    }
+  }, [ackPriceImpact, props.impact?.warning, showSpeedbump])
+
+  useEffect(() => {
+    if (showSpeedbump && props.impact?.warning !== 'error') {
+      setShowSpeedbump(false)
+    }
+  }, [ackPriceImpact, props.impact, showSpeedbump])
```

Comment:
this `useEffect` is intended to catch price changes, right? i.e. catching the edge case of the `impact.warning` changing while the speedbump is visible?

Reply:
Yeah, if the impact price goes under the speedbump threshold, we hide it. One the otherhand, if the impact price goes from under the threshold to above the threshold, we wait to show the speedbump after the user has accepted the new price (hence why setShowSpeedbump(true) is in a useCallback)